src/VS.Mefx/Commands/MatchChecker.cs

```diff
@@ -0,0 +1,446 @@
+            {
+                this.Options.ErrorWriter.WriteLine(Strings.MissingExportMessage);
+                exportingFields.ForEach(field => this.Options.ErrorWriter.WriteLine(field));
+            }
+
+            consideringExports.Sort((first, second) => first.ExportingField.CompareTo(second.ExportingField));
+            List<ImporterStorer> consideringImports = new List<ImporterStorer>();
+
+            // Perform matching against all considering imports
+            foreach (var import in importPart.Imports)
+            {
+                var currentImportDefintion = import.ImportDefinition;
+                string importingField = importPart.Type.FullName!;
+                if (import.ImportingMember != null)
+                {
+                    importingField = import.ImportingMember.Name;
+                }
+
+                bool performMatching = importingFields == null || importingFields.Contains(importingField);
+                if (performMatching)
+                {
+                    consideringImports.Add(new ImporterStorer(currentImportDefintion, importingField));
+                    if (importingFields != null)
+                    {
+                        importingFields.Remove(importingField);
+                    }
+                }
+            }
+
+            // Print message about which importing fields couldn't be found
+            if (importingFields != null && importingFields.Count() > 0)
+            {
+                this.Options.ErrorWriter.WriteLine(Strings.MissingImportMessage);
+                importingFields.ForEach(field => this.Options.ErrorWriter.WriteLine(field));
+            }
+
+            // Once we have found all the exports and imports to consider perform matching on them
+            consideringImports.Sort((first, second) => first.ImportingField.CompareTo(second.ImportingField));
+            foreach (var import in consideringImports)
+            {
+                string matchPreview = string.Format(
+                    CultureInfo.CurrentCulture,
+                    Strings.ConsideringImportFormat,
+                    import.ImportingField);
+                this.Options.Writer.WriteLine("\n" + matchPreview);
+
+                this.PerformDefinitionChecking(import.Import, consideringExports);
+            }
+        }
+
+        private class ImporterStorer
+        {
+            public ImporterStorer(ImportDefinition import, string importingField)
+            {
+                this.Import = import;
+                this.ImportingField = importingField;
+            }
+
+            public ImportDefinition Import { get; private set; }
+
+            public string ImportingField { get; private set; }
+        }
+
+        private class PartExport
+        {
+            /// <summary>
+            /// Name of key to use when getting type of a given export.
+            /// </summary>
+            private static readonly string TypeKey = "ExportTypeIdentity";
```

Comment:
prefer `const` over `static readonly` for private members (and even some public ones).